machine_learning/vector_ops.hpp

```diff
@@ -0,0 +1,476 @@
+ * @tparam T typename of the vector
+ * @param A 1D vector from which first element will be removed
+ * @return new resultant vector
+ */
+template <typename T>
+std::valarray <T> pop_front(const std::valarray<T> &A) {
+    std::valarray <T> B; // New 1D vector to store resultant vector
+    B.resize(A.size() - 1); // Resizing it accordingly
+    for(size_t i = 1; i < A.size(); i ++) { // // For every (except first) element in A 
+        B[i - 1] = A[i]; // Copy element in B with left shifted position
+    }
+    return B; // Return resultant vector
+}
+
+/**
+ * Function to remove last element from 1D vector
+ * @tparam T typename of the vector
+ * @param A 1D vector from which last element will be removed
+ * @return new resultant vector
+ */
+template <typename T>
+std::valarray <T> pop_back(const std::valarray<T> &A) {
+    std::valarray <T> B; // New 1D vector to store resultant vector
+    B.resize(A.size() - 1); // Resizing it accordingly
+    for(size_t i = 0; i < A.size() - 1; i ++) { // For every (except last) element in A 
+        B[i] = A[i]; // Copy element in B
+    }
+    return B; // Return resultant vector
+}
+
+/**
+ * Function to equally shuffle two 3D vectors (used for shuffling training data)
+ * @tparam T typename of the vector
+ * @param A First 3D vector
+ * @param B Second 3D vector
+ */
+template <typename T>
+void equal_shuffle(std::vector < std::vector <std::valarray<T>> >  &A, 
+                   std::vector < std::vector <std::valarray<T>> >  &B) {
+    // If two vectors have different sizes
+    if(A.size() != B.size())
+    {
+        std::cout << "ERROR : Can not equally shuffle two vectors with different sizes: ";
+        std::cout << A.size() << " and " << B.size() << std::endl;
+        std::exit(EXIT_FAILURE);
+    }
+    for(size_t i = 0; i < A.size(); i++) { // For every element in A and B
+        // Genrating random index < size of A and B
+        std::srand(std::chrono::system_clock::now().time_since_epoch().count());
+        size_t random_index = std::rand() % A.size();
+        // Swap elements in both A and B with same random index
+        std::swap(A[i], A[random_index]);
+        std::swap(B[i], B[random_index]);
+    }
+    return;
+}
+
+/**
+ * Function to initialize given 2D vector using uniform random initialization
+ * @tparam T typename of the vector
+ * @param A 2D vector to be initialized
+ * @param shape required shape
+ * @param low lower limit on value
+ * @param high upper limit on value
+ */
+template <typename T>
+void uniform_random_initialization(std::vector<std::valarray<T>> &A, 
+                   const std::pair<size_t, size_t> &shape, 
+                   const T &low, 
+                   const T &high) {
+    A.clear(); // Making A empty 
+    // Uniform distribution in range [low, high]
+    std::default_random_engine generator (std::chrono::system_clock::now().time_since_epoch().count());
+    std::uniform_real_distribution <T> distribution(low, high);
```

Comment:
```suggestion
    std::uniform_int_distribution <T> distribution(low, high);
```

Reply:
Actually i want real number's distribution.

Changing it gives me error:
```
C:/TDM-GCC-64/lib/gcc/x86_64-w64-mingw32/5.1.0/include/c++/bits/random.h: In instantiation of 'class std::uniform_int_distribution<double>':
vector_ops.hpp:161:39:   required from 'void uniform_random_initialization(std::vector<std::valarray<_Tp> >&, const std::pair<long long unsigned int, long long unsigned int>&, const 
T&, const T&) [with T = double]'
nn.hpp:173:90:   required from here
C:/TDM-GCC-64/lib/gcc/x86_64-w64-mingw32/5.1.0/include/c++/bits/random.h:1667:7: error: static assertion failed: template argument not an integral type
       static_assert(std::is_integral<_IntType>::value,
```

Reply:
It is fixed now with the `random` library.
Was just going to request adding it, thanks. 😄

Reply:
Including  ```<random>``` solved the problem